android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/DVCLifecycleCallbacks.kt

```diff
@@ -0,0 +1,70 @@
+package com.devcycle.sdk.android.eventsource
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import android.os.Handler
+import android.os.HandlerThread
+import timber.log.Timber
+
+class DVCLifecycleCallbacks(eventSource: EventSource, customHandler: Handler? = null) : Application.ActivityLifecycleCallbacks {
+    private val CHECK_DELAY: Long = 800
+
+    private val eventSource = eventSource
+    private val listenerThread: HandlerThread = HandlerThread("DVCForegroundListener")
+    private var handler: Handler? = customHandler
+    private var paused = false
+    private var foreground = true
+    private var check: Runnable? = null
+
+    init {
+        if (handler == null) {
+            listenerThread.start()
+            handler = Handler(listenerThread.looper)
+        }
+    }
+    override fun onActivityResumed(activity: Activity) {
+        paused = false
+        if (check != null) {
+            handler!!.removeCallbacks(check!!) // stop check from running if the timer is still running
+            check = null
+        }
+        if (!foreground) {
+            Timber.d("Restarting connection")
+            eventSource.start()
+        }
+        foreground = true
+    }
+
+    override fun onActivityPaused(activity: Activity) {
+        paused = true
+
+        if (check != null) {
+            handler!!.removeCallbacks(check!!)
+            check = null
+        }
+
+        check = Runnable {
+            if (foreground && paused) {
+                foreground = false
+                Timber.d("App backgrounded for $CHECK_DELAY ms, closing connection")
+                eventSource.close()
+            } else {
+                Timber.d("App paused but still foreground")
+            }
+        }
+
+        handler!!.postDelayed(check!!, CHECK_DELAY)
```

Comment:
we should be also checking if there's an activity delay in the config and use that instead of the hardcoded value https://github.com/DevCycleHQ/js-sdks/blob/cbbf28f81bd90ad53775ef985e3e39aa2782b4c9/sdk/js/src/Client.ts#L327